es-app/src/SystemScreenSaver.cpp

```diff
@@ -109,15 +117,87 @@ void SystemScreenSaver::startScreenSaver()
 			return;
 		}
 	}
+	else if (Settings::getInstance()->getString("ScreenSaverBehavior") == "slideshow")
+	{
+		// Configure to fade out the windows, Skip Fading if Instant mode
+		mState =  PowerSaver::getMode() == PowerSaver::INSTANT
+					? STATE_SCREENSAVER_ACTIVE
+					: STATE_FADE_OUT_WINDOW;
+		mVideoChangeTime = Settings::getInstance()->getInt("ScreenSaverSwapImageTimeout");
+		mOpacity = 0.0f;
+
+		// Load a random video
+		std::string path = "";
+		std::string img_src = Settings::getInstance()->getString("SlideshowScreenSaverImageSource");
+		if (img_src == "game list")
+		{
+			pickRandomGameListImage(path);
+		}
+		else if (img_src == "custom")
+		{
+			pickRandomCustomImage(path);
+			// Custom images are not tied to the game list
+			mCurrentGame = NULL;
+		}
+		else
+		{
+			LOG(LogError) << "Slideshow Screensaver - Invalid image source: " << img_src << "\n";
+		}
+
+		if (!mImageScreensaver)
+		{
+			mImageScreensaver = new ImageComponent(mWindow, false, false);
+		}
+
+		mTimer = 0;
+
+		mImageScreensaver->setImage(path);
+		mImageScreensaver->setOrigin(0.5f, 0.5f);
+		mImageScreensaver->setPosition(Renderer::getScreenWidth()/2, Renderer::getScreenHeight()/2);
+
+		if (Settings::getInstance()->getBool("SlideshowScreenSaverStretch"))
+		{
+			mImageScreensaver->setResize((float)Renderer::getScreenWidth(), (float)Renderer::getScreenHeight());
+		}
+		else
+		{
+			mImageScreensaver->setMaxSize((float)Renderer::getScreenWidth(), (float)Renderer::getScreenHeight());
+		}
+
+		if ((!mBackgroundAudio) && (Settings::getInstance()->getString("SlideshowScreenSaverBackgroundAudioFile") != ""))
+		{
+			if (boost::filesystem::exists(Settings::getInstance()->getString("SlideshowScreenSaverBackgroundAudioFile")))
+			{
+				mBackgroundAudio = Sound::get(Settings::getInstance()->getString("SlideshowScreenSaverBackgroundAudioFile"));
+				mBackgroundAudio->play();
```

Comment:
You should explicitly disable PS if background audio will be played. If wakeup time is set to  > PS mode then PS will engage even while playing images and halting the audio.